hip_data_tools/aws/athena.py

```diff
@@ -403,28 +403,46 @@ def explain_query(self):
 
         explain_query = f"Explain \n {self.query}"
 
-        results_dict = \
-            self.athena_util.run_query(explain_query, return_result=True)
+        results_dict = self.athena_util.run_query(explain_query, return_result=True)
 
         self.query_explaination = results_dict
 
     def extract_tables_from_explaination(self):
         """
-        Extracts the table references from the explaination and deposits the
+        Extracts unique table references from the explanation and deposits the
         table references in self.table_schema_entries
 
         Returns: None
 
         """
 
-        results_set = self.query_explaination['ResultSet']
+        results_set = self.query_explaination["ResultSet"]
+        explanation_rows = results_set["Rows"]
+        refs = [
+            self.parse_table_entries(data["VarCharValue"])
+            for row in explanation_rows
+            for data in row["Data"]
+        ]
+        all_table_refs = [item for sublist in refs for item in sublist]
+        self.table_schema_entries = self.extract_unique_table_references(all_table_refs)
 
-        explaination_rows = results_set['Rows']
+    @staticmethod
+    def extract_unique_table_references(all_table_refs: List[dict]) -> List[dict]:
```

Comment:
@shubhank-v As a practice we use functions instead of static methods. Let's make this static method a private function.